safenode/src/bin/safenode/rpc.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use safenode::node::NodeEventsChannel;
+
+use super::NodeCtrl;
+
+use eyre::{ErrReport, Result};
+use std::{env, net::SocketAddr, time::Duration};
+use tokio::sync::mpsc::{self, Sender};
+use tokio_stream::wrappers::ReceiverStream;
+use tonic::{transport::Server, Code, Request, Response, Status};
+use tracing::{debug, info, trace};
+
+use safenode_proto::safe_node_server::{SafeNode, SafeNodeServer};
+use safenode_proto::{
+    NodeEvent, NodeEventsRequest, NodeInfoRequest, NodeInfoResponse, RestartRequest,
+    RestartResponse, StopRequest, StopResponse, UpdateRequest, UpdateResponse,
+};
+
+// this would include code generated from .proto file
+#[allow(unused_qualifications, clippy::unwrap_used)]
+mod safenode_proto {
+    tonic::include_proto!("safenode_proto");
+}
+
+// Defining a struct to hold information used by our gRPC service backend
+struct SafeNodeRpcService {
+    addr: SocketAddr,
+    log_dir: String,
+    node_events_chnnel: NodeEventsChannel,
+    ctrl_tx: Sender<NodeCtrl>,
+}
+
+// Implementing RPC interface for service defined in .proto
+#[tonic::async_trait]
+impl SafeNode for SafeNodeRpcService {
+    type NodeEventsStream = ReceiverStream<Result<NodeEvent, Status>>;
+
+    async fn node_info(
+        &self,
+        request: Request<NodeInfoRequest>,
+    ) -> Result<Response<NodeInfoResponse>, Status> {
+        trace!(
+            "RPC request received at {}: {:?}",
+            self.addr,
+            request.get_ref()
+        );
+        let resp = Response::new(NodeInfoResponse {
+            node_name: xor_name::XorName::default().0.to_vec(),
```

Comment:
this will make all the NodeInfoResponse response with the same name
is it expected?
if so, maybe this means it can be removed actually?

Reply:
It's just because this PR is in draft yet, so not ready for review, but soon